syntax = "proto3";

service LogEventService {
  rpc LogTrade (TradeEvent) returns (LogAck);
  rpc LogOrder (OrderEvent) returns (LogAck);
  rpc LogFunding (FundingEvent) returns (LogAck);
  rpc LogLiquidation (LiquidationEvent) returns (LogAck);
}

message TradeEvent {
  string symbol = 1;
  string makerOrderId = 2;
  string takerOrderId = 3;
  double price = 4;
  double quantity = 5;
  string timestamp = 6;
}

message OrderEvent {
  string orderId = 1;
  string userId = 2;
  string symbol = 3;
  string side = 4;
  string type_detail = 5;
  double price = 6;
  double quantity = 7;
  string status = 8;
  string timestamp = 9;
}

message FundingEvent {
  string symbol = 1;
  double fundingRate = 2;
  double longOi = 3;
  double shortOi = 4;
  string timestamp = 5;
}

message LiquidationEvent {
  string userId = 1;
  string symbol = 2;
  double pnl = 3;
  string reason = 4;
  string timestamp = 5;
}

message LogAck {
  bool success = 1;
  string message = 2;
}
